cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -365,3 +365,121 @@ Given('device is rebooted', () => {
     throw new Error(CONSTANTS.STEP_IMPLEMENTATION_MISSING);
   });
 });
+
+/**
+ * @module fireboltCalls
+ * @function And 3rd party '(.+)' app is dismissed
+ * @description To dismiss the launched app
+ * @param {String} app - app name.
+ * @example
+ * And 3rd party 'firebolt' app is dismissed
+ */
+Given(/3rd party '(.+)' app is dismissed$/, async (appType) => {
+  const appId = Cypress.env(CONSTANTS.RUNTIME).appId;
+  let KeyPressSequence;
+  if (
+    Cypress.env(CONSTANTS.RUNTIME) &&
+    Cypress.env(CONSTANTS.RUNTIME).intent &&
+    Cypress.env(CONSTANTS.RUNTIME).intent.keyPressSequence
+  ) {
+    KeyPressSequence = Cypress.env(CONSTANTS.RUNTIME).intent.keyPressSequence;
+  } else if (
+    Cypress.env('app_metadata') &&
+    Cypress.env('app_metadata')[appId] &&
+    Cypress.env('app_metadata')[appId].defaultKeyPressSequence
+  ) {
+    KeyPressSequence = Cypress.env('app_metadata')[appId].defaultKeyPressSequence;
+  } else if (Cypress.env('app_metadata') && Cypress.env('app_metadata').defaultKeyPressSequence) {
+    KeyPressSequence = Cypress.env('app_metadata').defaultKeyPressSequence;
+  } else {
+    throw new Error(`Expected KeyPressSequence was not found for ${appId} in app_metadata.json`);
+  }
+  const requestMap = {
+    method: CONSTANTS.REQUEST_OVERRIDE_CALLS.DISMISS,
+    params: KeyPressSequence.dismiss,
+  };
+
+  fireLog.info(
+    `Request sent to platform to dismiss the ${appId} app: ${JSON.stringify(requestMap)}`
+  );
+  cy.sendMessagetoPlatforms(requestMap).then((response) => {
+    fireLog.info(`Response from platform: ${JSON.stringify(response)}`);
+  });
+});
+
+/**
+ * @module fireboltCalls
+ * @function And 3rd party '(.+)' app should be exited
+ * @description To validate that the app is dismissed
+ * @param {String} app - app name.
+ * @example
+ * Then 3rd party 'firebolt' app should be exited
+ */
+Given(/3rd party '(.+)' app should be exited$/, async (app) => {
+  // getAppState validation
+  const appId = Cypress.env(CONSTANTS.CURRENT_APP_ID);
+  const requestMapForGetAppState = {
+    method: CONSTANTS.REQUEST_OVERRIDE_CALLS.GETAPPSTATE,
+    params: appId,
+  };
+  fireLog.info(
+    `Sending request to fetch ${appId} app state: ${JSON.stringify(requestMapForGetAppState)}`
+  );
+  cy.sendMessagetoPlatforms(requestMapForGetAppState)
+    .then((response) => {
+      const responseString = JSON.stringify(response);
+      if (Object.keys(response).length === 0) {
+        fireLog.info(
+          `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+        );
+      } else {
+        fireLog.fail(`${appId} app is not dismissed. Response :${responseString}`);
+      }
+    })
+    .then(() => {
+      // screenShot validation
```

Comment:
when is the firebolt interactions enabled when executing deeplink ? like when does it start collection

Reply:
in the beforeAll
![image](https://github.com/user-attachments/assets/6a3b5c6f-f9e9-4af5-af44-f412e317fb4a)